lib/Doctrine/ORM/EntityManager.php

```diff
@@ -541,8 +541,13 @@ public function getPartialReference($entityName, $id)
         }
 
         foreach ($id as $i => $value) {
-            if (is_object($value) && $this->metadataFactory->hasMetadataFor(StaticClassNameConverter::getClass($value))) {
-                $id[$i] = $this->unitOfWork->getSingleIdentifierValue($value);
+            if (is_object($value)) {
+                try{
+                    $this->metadatas->get(StaticClassNameConverter::getClass($value));
+                    $id[$i] = $this->unitOfWork->getSingleIdentifierValue($value);
+                } catch (\Exception $e) {
```

Comment:
Likely wrong here - why did you introduce these try-catches? I know you are still playing around with the patch: just wondering why this got added to make things work

Reply:
just because for now get throws an Exception if the metadata is not found where it wasn't before.